pkgs/games/system-syzygy/default.nix

```diff
@@ -0,0 +1,44 @@
+{stdenv, rustPlatform, fetchFromGitHub, fetchurl, SDL2, makeWrapper, makeDesktopItem}:
+
+let
+  desktopFile = makeDesktopItem {
+    name = "system-syzygy";
+    exec = "%bin%";
+    comment = "A puzzle game";
+    desktopName = "System Syzygy";
+    categories = "Game;";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "system-syzygy";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "mdsteele";
+    repo = "syzygy";
+    rev = "5ba148fed7aae14bf35108d7303e4194e8ffe5e8";
+    sha256 = "07mzwx8ql33q865snnw4gm3dgf0mnm60lnq1f5fgas2yjy9g9vwa";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ SDL2 ];
+
+  cargoSha256 = "03724z33dqxbbrzpvz172qh45qrgnyb801algjdcm32v8xsx81qg";
+
+  postInstall = ''
+    mkdir -p $out/share/syzygy/
+    cp -r ${src}/data/* $out/share/syzygy/
+    wrapProgram $out/bin/syzygy --set SYZYGY_DATA_DIR $out/share/syzygy
+    mkdir -p $out/share/applications
+    substitute ${desktopFile}/share/applications/system-syzygy.desktop $out/share/applications/system-syzygy.desktop \
```

Comment:
Is there no other way to make a desktop file. Substituting feels like a bit of a hack

Reply:
I don't think it is possible to do this in another way to do this without breaking purity...

Reply:
Wouldn't `exec = "syzygy"` be sufficient?

Reply:
But I'll break purity, as it'll suppose that syzygy is in the path (I suppose)

Reply:
While it is true that it will expect it to be in the path I can't see a situation where the desktop item is exposed but the program is not in your path

Reply:
If you are set on this, perhaps `substituteAll` would be neater